exchanges/gateio/gateio_wrapper.go

```diff
@@ -1846,34 +1832,17 @@ func (g *Gateio) GetHistoricCandlesExtended(ctx context.Context, pair currency.P
 	if err != nil {
 		return nil, err
 	}
-	err = g.ValidateKline(pair, a, interval)
-	if err != nil {
-		return nil, err
-	}
-	var formattedPair currency.Pair
-	formattedPair, err = g.FormatExchangeCurrency(pair, a)
-	if err != nil {
-		return nil, err
-	}
 	var dates *kline.IntervalRangeHolder
-	dates, err = kline.CalculateCandleDateRanges(start, end, interval, 0)
+	dates, err = req.GetRanges(0)
```

Comment:
date ranges are included in the klineRequestExtended type, so this can be removed. Please replicate across all open PR's.